tower-balance/src/weight.rs

```diff
@@ -0,0 +1,167 @@
+use futures::{Async, Poll};
+use std::ops;
+use tower_discover::{Change, Discover};
+use tower_service::Service;
+
+use Load;
+
+/// A weight on [0.0, ∞].
+///
+/// Lesser-weighted nodes receive less traffic than heavier-weighted nodes.
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+pub struct Weight(f64);
+
+/// A Service, that implements Load, that
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub struct Weighted<T> {
+    inner: T,
+    weight: Weight,
+}
+
+#[derive(Debug)]
+pub struct WithWeighted<T>(T);
+
+pub trait HasWeight {
```

Comment:
This might be overly nitpicky, but the distinction between `HasWeight` and `Weighted` is not particularly clear to me from the names...I might have called this trait `Weighted` and the service type `WeightedService`...TIOLI of course.